File: JAVA/MostFrequentDaysTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertArrayEquals;
import org.junit.Test;

public class MostFrequentDaysTest {
    
    @Test
	public void testMostFrequentDays() throws Exception {
		//Test 1
		String[] want = {"Friday"};
		String[] got = MostFrequentDays.mostFrequentDays( 2427 );
        assertArrayEquals(want, got);
		//Test 2
		want = new String[] {"Saturday"};
		got = MostFrequentDays.mostFrequentDays( 2185 );
        assertArrayEquals(want, got);
		//Test 3
		want = new String[] {"Tuesday", "Wednesday"};
		got = MostFrequentDays.mostFrequentDays( 1084 );
        assertArrayEquals(want, got);
		//Test 4
		want = new String[] {"Sunday"};
		got = MostFrequentDays.mostFrequentDays( 1167 );
        assertArrayEquals(want, got);
		//Test 5
		want = new String[] {"Friday", "Saturday"};
		got = MostFrequentDays.mostFrequentDays( 1216 );
        assertArrayEquals(want, got);
		//Test 6
		want = new String[] {"Friday", "Saturday"};
		got = MostFrequentDays.mostFrequentDays( 1492 );
        assertArrayEquals(want, got);
		//Test 7
		want = new String[] {"Monday"};
		got = MostFrequentDays.mostFrequentDays( 1770 );
        assertArrayEquals(want, got);
		//Test 8
		want = new String[] {"Saturday"};
		got = MostFrequentDays.mostFrequentDays( 1785 );
        assertArrayEquals(want, got);
		//Test 9
		want = new String[] {"Wednesday", "Thursday"};
		got = MostFrequentDays.mostFrequentDays( 212 );
        assertArrayEquals(want, got);
	}

}

File: JS/MostFrequentDays.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./MostFrequentDays.js')

TEST('Starting MostFrequentDays test...', (t) => {
    //Test 1
    let want = ["Friday"]
    let got = OBJ.mostFrequentDays( 2427 )
    t.assert(want.toString === got.toString, "Expect: 2427 = Friday")
    //Test 2
    want = ["Saturday"]
    got = OBJ.mostFrequentDays( 2185 )
    t.assert(want.toString === got.toString, "Expect: 2185 = Saturday")
    //Test 3
    want = ["Tuesday", "Wednesday"]
    got = OBJ.mostFrequentDays( 1084 )
    t.assert(want.toString === got.toString, "Expect: 1084 = Tuesday, Wednesday")
    //Test 4
    want = ["Sunday"]
    got = OBJ.mostFrequentDays( 1167 )
    t.assert(want.toString === got.toString, "Expect: 1167 = Sunday")
    //Test 5
    want = ["Friday", "Saturday"]
    got = OBJ.mostFrequentDays( 1216 )
    t.assert(want.toString === got.toString, "Expect: 1216 = Friday, Saturday")
    //Test 6
    want = ["Friday", "Saturday"]
    got = OBJ.mostFrequentDays( 1492 )
    t.assert(want.toString === got.toString, "Expect: 1492 = Friday, Saturday")
    //Test 7
    want = ["Monday"]
    got = OBJ.mostFrequentDays( 1770 )
    t.assert(want.toString === got.toString, "Expect: 1770 = Monday")
    //Test 8
    want = ["Saturday"]
    got = OBJ.mostFrequentDays( 1785 )
    t.assert(want.toString === got.toString, "Expect: 1785 = Saturday")
    //Test 9
    want = ["Wednesday", "Thursday"]
    got = OBJ.mostFrequentDays( 212 )
    t.assert(want.toString === got.toString, "Expect: 212 = Wednesday, Thursday")

    t.end()
})
